plugins/rake-fast/rake-fast.plugin.zsh

```diff
@@ -8,7 +8,19 @@ _rake_refresh () {
 }
 
 _rake_does_task_list_need_generating () {
-  [[ ! -f .rake_tasks ]] || [[ Rakefile -nt .rake_tasks ]]
+  _rake_tasks_missing || _rakefile_has_changes || _sub_rake_files_have_changes
+}
+
+_rake_tasks_missing () {
+  [[ ! -f .rake_tasks ]]
+}
+
+_rakefile_has_changes () {
+  [[ Rakefile -nt .rake_tasks ]]
+}
+
+_sub_rake_files_have_changes () {
+  [[ $(find ./**/*.rake(.om[1])) -nt .rake_tasks ]]
```

Comment:
Sure. I figured it should look at any *.rake file recursively in the current directory. Just in case there are nested folders within the lib/tasks directory or if rake files are defined elsewhere.

So the `./**/*.rake` part gets all files with the .rake extension anywhere in the current directory. Then in the parentheses, `.` specifies it must be a file (as opposed to a directory), `om` orders the results by modified at time, and `[1]` returns the first result (the most recently modified file). 

I used http://reasoniamhere.com/2014/01/11/outrageously-useful-tips-to-master-your-z-shell/ as a reference for the parts in the parentheses.